modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultConstantFoldingOptimizationPhase.java

```diff
@@ -843,11 +843,11 @@ private void replaceCallWithConstant(
             ExpressionNode argNode = irInvokeCallMemberNode.getArgumentNodes().get(i);
             IRDConstant constantDecoration = argNode.getDecoration(IRDConstant.class);
             if (constantDecoration == null) {
-                // TODO find a better string to output
                 throw irInvokeCallMemberNode.getLocation()
                     .createError(
                         new IllegalArgumentException(
-                            "all arguments to [" + javaMethod.getName() + "] must be constant but the [" + (i + 1) + "] argument isn't"
+                            "The [" + javaMethod.getName() + "] method is sad because it needs constant arguments to work properly. " +
+                                "Please provide a constant to the [" + (i + 1) + "] argument"
```

Comment:
I suppose it's kinder, but I was really looking for something different. I'm not totally sure what - that's why I was vague in the issue. I think my concerns were more around things like "if there is a single argument this sentence sounds silly" and "maybe we should use ordinal numbers like `second` instead of `[2]`". I don't think being polite is really important here.

Reply:
Thanks for the feedback @nik9000! I had the concern to try to reduce the negative impact of the user error according to the Guidelines i mentioned above, but since this is not the priority, i've just updated the error messages to display ordinal numbers and to be more direct to the point.